lib/shopify_transporter/pipeline/magento/order/top_level_attributes.rb

```diff
@@ -25,13 +34,79 @@ def convert(hash, record)
 
           private
 
+          ORDER_ORIGINATED_FROM = 'Magento'
+
           def build_customer(hash)
             {
               email: hash['customer_email'],
               first_name: hash['customer_firstname'],
               last_name: hash['customer_lastname'],
             }.stringify_keys
           end
+
+          def financial_status(hash)
+            total_price = hash['grand_total'].to_i
+            total_paid = hash['total_paid'].present? ? hash['total_paid'].to_i : 0
+            total_refunded = hash['total_refunded'].present? ? hash['total_refunded'].to_i : 0
+            order_state = hash['state']
+            status = nil
+
+            if order_state == 'Pending Payment'
```

Comment:
wow

Reply:
yep i had a similar conversation with ziyan

the code is pretty much as good as it's possible to be. it's unfortunate.

Reply:
one tiny thing that might be an improvement (or might not, so give me your honest opinion):

change 

```
elsif total_price == total_paid && total_refunded == 0
              status = 'paid'
```

to

```
elsif paid?(hash)
              status = 'paid'
```

this allows you to move the logic from lines 48 to 51 into multiple small helper functions instead of at the top of this one, and allows you to make the big ugly section from lines 54 to 64 much more readable.

on the other hand, it means you're reading the contents of the hash in a place that is more separated from where that content matters, which is an extra layer of indirection that maybe isn't providing value. thoughts?

Reply:
+1 on this. For the purposes of readability, it might be better to actually wrap each of these conditions in the financial status name as a method: `paid?` `partially_paid?` etc. This will help with readability in the long run.

Reply:
Thanks for the input! I'll extract these conditions into small methods.